Yodel/Functions/Invoke-YSqlServerCommand.ps1

```diff
@@ -0,0 +1,178 @@
+
+function Invoke-YSqlServerCommand
+{
+    <#
+    .SYNOPSIS
+    Uses ADO.NET to execute a query in a SQL Server database.
+
+    .DESCRIPTION
+    `Invoke-YSqlServerCommand` executes a SQL query against a SQL Server database. The function opens a connection to SQL Server, executes the query, then closes the connection. Pass the name of the SQL Server (hostname and instance name) to the `SqlServerName` parameter. Pass the database name to the `DatabaseName` parameter.  By default, the query is run as the current user. To run as a custom user, pass the user's credentials to the `Credential` parameter. 
+    
+    Pass the query to run to the `Text` parameter. You may also pipe queries to `Invoke-YSqlServerCommand`. Piped queries are all run using the same connection. 
+    
+    The function returns a generic object for each row in the result set. Each object has a property for each column. If a column doesn't have a name, a generic `ColumnX` name is assigned, where `X` starts at 0 and increases by one for each nameless column.
+
+    If your query returns a single value, use the `-AsScalar` switch.
+
+    If your query returns no results, use the `-NonQuery` switch. If your query affects any rows, the number of rows affected will be returned.
+
+    To run a parameterized query, use `@name` parameters in your query and pass the values of those parameters in a hashtable to the `Parameter` parameter, e.g. `@{ '@name' = 'the_name' }`.
+
+    To execute a stored procedure, set the `Text` parameter to the name of the stored procedure, set the `Type` parameter to `[Data.CommandType]::StoredProcedure`, and pass the procedure's parameters to the `Parameter` parameter (a hashtable of parameter names and values).
+
+    Queries will time out after 30 seconds (the default .NET timeout). If you have a query that runs longer, pass the number of seconds to wait to the `Timeout` parameter. 
+
+    Query timings are output to the verbose stream, including the text of the query. If you want to suppress sensitive queries from being output, set the `Verbose` parameter to `$false`, e.g. `-Verbose:$false`.

```

Comment:
The thought crossed my mind. I stuck with `-Verbose` because that's what is being controlled: the verbose message. If we did anything else with the query, I think a `-Sensitive` switch would make sense.